x-pack/plugins/security_solution/public/detection_engine/rule_gaps/components/manual_rule_run/index.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiBetaBadge,
+  EuiConfirmModal,
+  EuiDatePicker,
+  EuiDatePickerRange,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiHorizontalRule,
+  EuiSpacer,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import moment from 'moment';
+import React, { useCallback, useMemo, useState } from 'react';
+import { TECHNICAL_PREVIEW, TECHNICAL_PREVIEW_TOOLTIP } from '../../../../common/translations';
+
+import * as i18n from './translations';
+
+interface ManualRuleRunModalProps {
+  onCancel: () => void;
+  onConfirm: (timeRange: { startDate: moment.Moment; endDate: moment.Moment }) => void;
+}
+
+const ManualRuleRunModalComponent = ({ onCancel, onConfirm }: ManualRuleRunModalProps) => {
+  const modalTitleId = useGeneratedHtmlId();
+
+  const now = useMemo(() => moment(), []);
+
+  const [startDate, setStartDate] = useState(now.clone().subtract(1, 'd'));
+  const [endDate, setEndDate] = useState(now.clone());
+
+  const isEndDateInFuture = endDate.isAfter(now);
+  const isInvalid = isEndDateInFuture || startDate.isSameOrAfter(endDate);
```

Comment:
can we add validation for max date in past as on API side?

<img width="2300" alt="Screenshot 2024-06-06 at 12 37 16" src="https://github.com/elastic/kibana/assets/92328789/4fc70d8b-2121-41cd-a667-ccd4a088829c">

Reply:
good catch, I missed this error